portal/import_template_ui.php

```diff
@@ -15,303 +15,543 @@
-                                } else {
-                                    echo "<option value='" . text($dir['option_id']) . "'>" . text($dir['title']) . "</option>\n";
-                                }
-                            }
-                            ?>
-                        </select>
-                        <label class="label mx-1" for="sel_pt"><?php echo xlt('Patient'); ?></label>
-                        <select class="form-control" id="sel_pt" name="sel_pt">
-                            <option value='0'><?php echo xlt("All Patients") ?></option>
+                        <label class='font-weight-bold mx-1' for='selected_patients'><?php echo xlt('Location'); ?></label>
+                        <select class="form-control select-dropdown" id="selected_patients" name="selected_patients[]" multiple>
                             <?PHP
                             $ppt = getAuthUsers();
-                            global $patient_dir;
                             foreach ($ppt as $pt) {
-                                if ($patient_dir != $pt['pid'] . "_tpls") {
-                                    echo "<option value=" . attr($pt['pid']) . ">" . text($pt['ptname']) . "</option>";
+                                if ((is_array($patient) && !in_array($pt['pid'], $patient)) || empty($patient)) {
+                                    echo '<option value=' . attr($pt['pid']) . '>' . text($pt['ptname']) . '</option>';
                                 } else {
-                                    echo "<option value='" . attr($pt['pid']) . "' selected='selected'>" . text($pt['ptname']) . "</option>";
+                                    echo "<option value='" . attr($pt['pid']) . "' selected='selected'>" . text($pt['ptname'] . ' ') . '</option>';
                                 }
                             }
                             ?>
                         </select>
+                        <a class='btn-refresh ml-1' onclick="$('#selected_patients').val(null).trigger('change');" role="button"></a>
+                        <?php
+                        $select_cat_options = '<option value="">' . xlt('General') . "</option>\n";
+                        foreach ($category_list as $option_category) {
+                            if (stripos($option_category['option_id'], 'repository') !== false) {
+                                continue;
+                            }
+                            if ($category === $option_category['option_id']) {
+                                $select_cat_options .= "<option value='" . attr($option_category['option_id']) . "' selected>" . text($option_category['title']) . "</option>\n";
+                            } else {
+                                $select_cat_options .= "<option value='" . text($option_category['option_id']) . "'>" . text($option_category['title']) . "</option>\n";
```

Comment:
attr() in attribute
(i finally get to point out some escaping issues, been so long, almost brings a tear to my eye; you are still batting 99.9% on escaping, just a couple slipped through :) )

Reply:
Did ya ever think I may be just testing you Brady. Just to see if I can get away with anything! Yep, that's what I'll go with...:)

Reply:
Definitely makes sense. Just keeping me on my toes :)